src/main/java/gov/loc/repository/bagit/conformance/profile/BagitProfileDeserializer.java

```diff
@@ -36,149 +37,176 @@ public BagitProfileDeserializer(final Class<?> vc) {
 
   @Override
   public BagitProfile deserialize(final JsonParser p, final DeserializationContext ctxt)
-      throws IOException, JsonProcessingException {
+          throws IOException, JsonProcessingException {
     final BagitProfile profile = new BagitProfile();
     final JsonNode node = p.getCodec().readTree(p);
-    
+
     parseBagitProfileInfo(node, profile);
-    
+
     profile.setBagInfoRequirements(parseBagInfo(node));
-    
+
     profile.getManifestTypesRequired().addAll(parseManifestTypesRequired(node));
-    
+
     profile.setFetchFileAllowed(node.get("Allow-Fetch.txt").asBoolean());
     logger.debug(messages.getString("fetch_allowed"), profile.isFetchFileAllowed());
-    
+
     profile.setSerialization(Serialization.valueOf(node.get("Serialization").asText()));
-    logger.debug(messages.getString("serialization_allowed"),profile.getSerialization());
-    
+    logger.debug(messages.getString("serialization_allowed"), profile.getSerialization());
+
     profile.getAcceptableMIMESerializationTypes().addAll(parseAcceptableSerializationFormats(node));
-    
+
     profile.getTagManifestTypesRequired().addAll(parseRequiredTagmanifestTypes(node));
-    
+
     profile.getTagFilesRequired().addAll(parseRequiredTagFiles(node));
-    
+
     profile.getAcceptableBagitVersions().addAll(parseAcceptableVersions(node));
-    
+
     return profile;
   }
-  
-  private static void parseBagitProfileInfo(final JsonNode node, final BagitProfile profile){
+
+  private static void parseBagitProfileInfo(final JsonNode node, final BagitProfile profile) {
     final JsonNode bagitProfileInfoNode = node.get("BagIt-Profile-Info");
     logger.debug(messages.getString("parsing_bagit_profile_info_section"));
-    
+
+    // Read required tags first
+    // due to specification defined at https://github.com/bagit-profiles/bagit-profiles
```

Comment:
Not really sure what this comment means. Why do you need to read required tags first?

Reply:
First the required tags are read and then the optional tags because they have to be handled differently. It's just to keep the code clearer.
If one of the required ones do not already exist, the optional ones are no longer read at all.

Reply:
unless there is an error being thrown that I don't know about, how are you stopping if a required tag is missing? To put it another way, what would break if you moved those `if not null` statements to being read first?

Perhaps a better way to make it cleaner would be to make another method that contained all the reads for optional tags. Something like
```readOptionalTags(final JsonNode bagitProfileInfoNode, final BagitProfile profile){
    final JsonNode contactNameNode = bagitProfileInfoNode.get("Contact-Name");
    if (contactNameNode != null) {
      final String contactName = contactNameNode.asText();
      logger.debug(messages.getString("contact_name"), contactName);
      profile.setContactName(contactName);
    }
     final JsonNode contactEmailNode = bagitProfileInfoNode.get("Contact-Email");
    if (contactEmailNode != null) {
      final String contactEmail = contactEmailNode.asText();
      logger.debug(messages.getString("contact_email"), contactEmail);
      profile.setContactEmail(contactEmail);
    }
     final JsonNode contactPhoneNode = bagitProfileInfoNode.get("Contact-Phone");
    if (contactPhoneNode != null) {
      final String contactPhone = contactPhoneNode.asText();
      logger.debug(messages.getString("contact_phone"), contactPhone);
      profile.setContactPhone(contactPhone);
    }
}
```

Reply:
If one of the required tags is not present, a NullPointerException is thrown.
> final String profileIdentifier = bagitProfileInfoNode.get("BagIt-Profile-Identifier")**.asText()**;
See documentation of class BagLinter:
> This class is only to be used on VALID bags, using it on un-validated bags may result in
>  exceptions being thrown (like {@link java.io.IOException} )
Therefore, no error handling is done anywhere in the code.